.github/workflows/build_and_push.yml

```diff
@@ -0,0 +1,46 @@
+name: Renku Base Docker Image CI
+
+on: [push]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        RENKU_VERSION: ["", "0.8.2", "0.9.0"]
+        EXTENSIONS: ["base", "r", "r-bioconductor_3_10", "cuda-9.2", "cuda-10.0-tensorflow-1.14"]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build the Renku Docker pull request images for testing
+      if: github.event_name == 'pull_request'
+      run: |
+        docker build docker/${{ matrix.EXTENSIONS }} --tag $RENKU_DOCKER_TAG
+        docker push $RENKU_DOCKER_TAG
+      env:
+        GIT_HASH: $(git rev-parse --short "$GITHUB_SHA")
```

Comment:
There is a syntax in github actions you can use to set environment variables from a step that will be available to other steps. So you can have a step before this one that would do something like what is done here: https://github.com/SwissDataScienceCenter/renku/blob/ed178ffd84ff58ed9cc1a5e7c79d62e958ae2c09/.github/workflows/pull-request-test.yml#L52